store.go

```diff
@@ -3311,6 +3323,27 @@ func (s *store) ContainerByLayer(id string) (*Container, error) {
 	return nil, ErrContainerUnknown
 }
 
+func (s *store) ImageDirectory(id string) (string, error) {
+	foundImage := false
+	if res, done, err := readAllImageStores(s, func(store roImageStore) (string, bool, error) {
+		if store.Exists(id) {
+			foundImage = true
+		}
```

Comment:
This is surprising, is this right? Besides `foundImage` being dead, as noted elsewhere, the effect is that for images in an additional store, we create these data directories in the _primary_ store which doesn’t store the image at all.

That _might_ be right (assuming the additional stores are read-only and we simply can’t write them to additional stores — I don’t know what the use case is and whether we need to write this data to read-only images), but then we probably need some new image cleanup code to remove this data in the otherwise-unrelated store when the image is being deleted.

If that was not the intent, I’d expect this to look more like the image part of `store.Metadata`:
```go
if res, done, err := readAllImageStores … {
    if store.Exists(id) {
        res, err := …
        return res, true, err
     }
     return N/A, false, nil
}; done {
   return res, err
}
return "", “not found”
```

(Applies to both.)